src/input-multi-select/input-multi-select.tsx

```diff
@@ -70,7 +80,7 @@ export const InputMultiSelect = <T, V>({
     // EVENT HANDLERS
     // =============================================================================
     const handleSelectAllClick = () => {
-        if (selected && selected.length > 0) {
+        if ((selected && selected.length > 0) || isMultiSelectLimit) {
```

Comment:
what scenario do we need to check that user has selected the max? because if they have selected the max, then `selected.length` should also be >0